cl_dll/saytext.cpp

```diff
@@ -105,6 +106,10 @@ int CHudSayText :: Draw( float flTime )
 	if ( ( gViewPort && gViewPort->AllowedToPrintText() == FALSE) || !m_HUD_saytext->value )
 		return 1;
 
+	bool enableRainbow = m_HUD_rainbow_chat->value;
+	if (!enableRainbow)
+		gHUD.m_Rainbow.PushDisable();
```

Comment:
Is this recursive? I.e. why does this need a counter instead of a simple true/false?

Reply:
Futureproofing. With a true/false code like this will unexpectedly break:
```cpp
void SomeCommonFunc() {
  SetEnabled(false);
  // draw something
  SetEnabled(true);
}

void DrawSomeElement() {
  SetEnabled(false);
  // ...
  SomeCommonFunc();
  // Effect now enabled
  // ...
  SetEnabled(true);
}
```